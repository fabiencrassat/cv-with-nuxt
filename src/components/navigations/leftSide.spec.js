import { shallowMount } from '@vue/test-utils';
import NavigationLeftSide from './leftSide';

jest.mock('../icons/user', () => jest.fn);
jest.mock('../icons/phone', () => jest.fn);
jest.mock('../icons/glasses', () => jest.fn);
jest.mock('../icons/suitcase', () => jest.fn);

const factory = ({ propsData } = {}) => {
  return shallowMount(NavigationLeftSide, {
    mocks: {
      // Always returns the input
      $t: (i) => `$t('${i}')`,
      localePath: (i) => `localePath('${i}')`,
    },
    propsData,
    stubs: ['nuxt-link', 'AboutIcon', 'ContactIcon', 'SkillIcon', 'JobIcon'],
  });
};

describe('leftSide', () => {
  it('mounts properly', () => {
    expect.hasAssertions();
    const wrapper = factory();
    // eslint-disable-next-line jest/no-truthy-falsy
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('renders properly', () => {
    expect.hasAssertions();
    const wrapper = factory();
    // eslint-disable-next-line jest/prefer-inline-snapshots
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('renders properly with props', () => {
    expect.hasAssertions();
    const wrapper = factory({ propsData: { homepage: '/foo' } });
    // eslint-disable-next-line jest/prefer-inline-snapshots
    expect(wrapper.html()).toMatchSnapshot();
  });
});
