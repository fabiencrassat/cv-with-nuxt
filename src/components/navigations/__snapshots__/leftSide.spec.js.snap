// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`leftSide renders properly 1`] = `
<nav class="z-10 sticky top-0 bg-blue-700 text-gray-200">
  <div class="flex justify-between items-center py-1 sm:py-0 bg-gray-900 bg-opacity-25 text-center">
    <div id="menuToggle" class="block relative mx-2 text-left"><input type="checkbox" title="$t('menu')" class="block absolute cursor-pointer opacity-0 z-50 w-10 h-8 top-0 sm:hidden"> <span class="z-40 block relative bg-gray-200 w-8 h-1 my-1 rounded-sm transition-transform duration-500 ease-in-out sm:hidden"></span> <span class="z-40 block relative bg-gray-200 w-8 h-1 my-1 rounded-sm transition duration-500 ease-in-out sm:hidden"></span> <span class="z-40 block relative bg-gray-200 w-8 h-1 my-1 rounded-sm transition-transform duration-500 ease-in-out sm:hidden"></span>
      <div id="menu" class="z-20 fixed top-0 left-0 bg-blue-700 w-screen sm:w-24 md:w-56 h-screen transform -translate-x-full sm:translate-x-0 transition-transform duration-500 ease-in-out overflow-y-scroll">
        <header class="py-8 pl-2 pr-6 sm:p-2 md:py-8 md:px-2 text-center bg-gray-900 bg-opacity-25 sm:relative">
          <profilepicture-stub name="foo" picture="bar" class="overflow-hidden mx-auto sm:w-16 sm:h-16 md:w-24 md:h-24"></profilepicture-stub>
          <name-stub name="foo" class="pt-4"></name-stub>
          <switchlang-stub class="pt-4"></switchlang-stub>
        </header>
        <ul class="py-2"></ul>
      </div>
    </div>
    <profilepicture-stub name="foo" picture="bar" class="w-16 h-16 mx-1 sm:hidden"></profilepicture-stub>
    <name-stub name="foo" class="mx-1 sm:hidden"></name-stub>
    <switchlang-stub class="mx-1 sm:hidden"></switchlang-stub>
  </div>
</nav>
`;

exports[`leftSide renders properly with props 1`] = `
<nav class="z-10 sticky top-0 bg-blue-700 text-gray-200">
  <div class="flex justify-between items-center py-1 sm:py-0 bg-gray-900 bg-opacity-25 text-center">
    <div id="menuToggle" class="block relative mx-2 text-left"><input type="checkbox" title="$t('menu')" class="block absolute cursor-pointer opacity-0 z-50 w-10 h-8 top-0 sm:hidden"> <span class="z-40 block relative bg-gray-200 w-8 h-1 my-1 rounded-sm transition-transform duration-500 ease-in-out sm:hidden"></span> <span class="z-40 block relative bg-gray-200 w-8 h-1 my-1 rounded-sm transition duration-500 ease-in-out sm:hidden"></span> <span class="z-40 block relative bg-gray-200 w-8 h-1 my-1 rounded-sm transition-transform duration-500 ease-in-out sm:hidden"></span>
      <div id="menu" class="z-20 fixed top-0 left-0 bg-blue-700 w-screen sm:w-24 md:w-56 h-screen transform -translate-x-full sm:translate-x-0 transition-transform duration-500 ease-in-out overflow-y-scroll">
        <header class="py-8 pl-2 pr-6 sm:p-2 md:py-8 md:px-2 text-center bg-gray-900 bg-opacity-25 sm:relative">
          <profilepicture-stub name="foo" picture="bar" class="overflow-hidden mx-auto sm:w-16 sm:h-16 md:w-24 md:h-24"></profilepicture-stub>
          <name-stub name="foo" class="pt-4"></name-stub>
          <switchlang-stub class="pt-4"></switchlang-stub>
        </header>
        <ul class="py-2">
          <li>
            <nuxt-link-stub to="#page1" title="Page 1" class="link block sm:text-sm p-4 outline-none bg-no-repeat cursor-pointer hover:bg-gray-900 hover:bg-opacity-25 sm:text-center md:text-left">
              <component1-stub class="h-5 w-5 sm:h-8 sm:w-8 md:h-5 md:w-5"></component1-stub> <span class="ml-3 capitalize sm:hidden md:inline">
                Page 1
              </span>
            </nuxt-link-stub>
          </li>
          <li>
            <nuxt-link-stub to="#page2" title="Page 2" class="link block sm:text-sm p-4 outline-none bg-no-repeat cursor-pointer hover:bg-gray-900 hover:bg-opacity-25 sm:text-center md:text-left">
              <component2-stub class="h-5 w-5 sm:h-8 sm:w-8 md:h-5 md:w-5"></component2-stub> <span class="ml-3 capitalize sm:hidden md:inline">
                Page 2
              </span>
            </nuxt-link-stub>
          </li>
        </ul>
      </div>
    </div>
    <profilepicture-stub name="foo" picture="bar" class="w-16 h-16 mx-1 sm:hidden"></profilepicture-stub>
    <name-stub name="foo" class="mx-1 sm:hidden"></name-stub>
    <switchlang-stub class="mx-1 sm:hidden"></switchlang-stub>
  </div>
</nav>
`;
